src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useCallback, FunctionComponent } from 'react';
+
+import BoltCustomForm from './BoltCustomForm';
+import { HostedPaymentMethodProps } from './HostedPaymentMethod';
+import HostedWidgetPaymentMethod from './HostedWidgetPaymentMethod';
+
+const BoltEmbeddedPaymentMethod: FunctionComponent<HostedPaymentMethodProps> = ({
+    initializePayment,
+    method,
+    ...rest
+}) => {
+    const boltEmbeddedContainerId = 'bolt-embedded';
+
+    const initializeBoltPayment = useCallback(options => initializePayment({
+            ...options,
+            bolt: {
+                containerId: boltEmbeddedContainerId,
+                useBigCommerceCheckout: true,
+            },
+        }
+    ), [initializePayment, boltEmbeddedContainerId]);
+
+    function renderCustomPaymentForm() {
```

Comment:
Can we define it as `useCallback` so that the prop is memoized?

Reply:
Ok I will define it as `useCallback`

Reply:
Changed